internal/ingest/linksystem.go

```diff
@@ -135,13 +134,35 @@ func verifyAdvertisement(n ipld.Node, reg *registry.Registry) (peer.ID, error) {
 // indexer.
 func (ing *Ingester) ingestAd(publisherID peer.ID, adCid cid.Cid, ad schema.Advertisement) error {
 	stats.Record(context.Background(), metrics.IngestChange.M(1))
+	var mhCount int
+	var entsSyncStart time.Time
+	var entsStoreElapsed time.Duration
 	ingestStart := time.Now()
-	defer func() {
-		stats.Record(context.Background(), metrics.AdIngestLatency.M(coremetrics.MsecSince(ingestStart)))
-	}()
 
 	log := log.With("publisher", publisherID, "adCid", adCid)
 
+	defer func() {
+		now := time.Now()
+
+		// Record how long ad sync took.
+		elapsed := now.Sub(ingestStart)
+		elapsedMsec := float64(elapsed.Nanoseconds()) / 1e6
+		stats.Record(context.Background(), metrics.AdIngestLatency.M(elapsedMsec))
+		log.Infow("Finished syncing advertisement", "elapsed", elapsed.String(), "multihashes", mhCount)
+
+		if mhCount == 0 {
+			return
+		}
+		// Record how long entries sync took.
+		elapsed = now.Sub(entsSyncStart)
+		elapsedMsec = float64(elapsed.Nanoseconds()) / 1e6
+		stats.Record(context.Background(), metrics.EntriesSyncLatency.M(elapsedMsec))
+
+		// Record average time to store 10000 multihashes.
+		elapsedPer10KMh := float64(entsStoreElapsed.Nanoseconds()/int64(mhCount)) / 100
```

Comment:
the `/100` is incorporating both the `ns-ms` conversion and the `10k` batching, right? maybe break those two factors out for clarity?